enhancements/update/upgrade-blocker-operator.md

```diff
@@ -0,0 +1,128 @@
+---
+title: upgrade-blocker-operator
+authors:
+  - "@michaelguigno"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-03
+last-updated: 2021-02-03
+status: implementable
+---
+
+# Upgrade Blocker Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Add an extensible API and associated operator to block upgrades.
+
+## Motivation
+
+Sometimes clusters shouldn't be upgraded.  Currently, the only signal is
+ClusterOperator statuses monitored by the CVO.  Cluster administrators need
+to be able to optionally inform the cluster when upgrades aren't appropriate.
```

Comment:
Naive question, but doesn't upgrade require an action from a cluster administrator?  So is this enhancement about adding a pluggable model that would allow administrators to encode arbitrary pre-upgrade checks into the tool itself, rather than having the administrator manually do checks before triggering the update?

Reply:
If you're managing a fleet of say, a dozen clusters, you don't necessarily want to stop and ensure each one meets some arbitrary criteria, or worse, start an upgrade when the criteria has not been met.

Also, the end goal is to have cluster upgrades be automatic.  Having an API like this would potentially allow someone to turn on automatic upgrades, and still have a large degree over when and whether those upgrades take place.